src/extension/tab/hook.js

```diff
@@ -70,28 +13,63 @@ export function installHook(window, devToolsVersion) {
     },
   });
 
-  // XXX this is a patch to back support previous versions of Apollo Client
-  // at somepoint we should remove this.
-  // the newer version has the client connecting to the hook, not the other
-  // way around that it currently does
-  let interval;
-  let count = 0;
+  function handleActionHookForDevtools() {
+    window.postMessage({
+      message: 'action-hook-fired',
+      to: 'tab:background:devtools',
+    });
+  }
+  
   function findClient() {
-    // only try for 10seconds
-    if (count++ > 10) clearInterval(interval);
-    if (!!window.__APOLLO_CLIENT__) {
-      hook.ApolloClient = window.__APOLLO_CLIENT__;
-      hook.ApolloClient.__actionHookForDevTools(
-        ({
-          state: { queries, mutations },
-          dataWithOptimisticResults: inspector,
-        }) => {
-          hook.actionLog.push({ queries, mutations, inspector });
-        },
-      );
-      clearInterval(interval);
+    let interval;
+    let count = 0;
+  
+    function initializeDevtoolsHook() {
+      if (count++ > 10) clearInterval(interval);
+      if (!!window.__APOLLO_CLIENT__) {
+        hook.ApolloClient = window.__APOLLO_CLIENT__;
+        hook.ApolloClient.__actionHookForDevTools(handleActionHookForDevtools);
+        hook.getQueries = () => hook.ApolloClient.queryManager.queries();
+        hook.getMutations = () => hook.ApolloClient.queryManager.mutationStore.getStore();
+        hook.getCache = () => hook.ApolloClient.cache.extract(true);
```

Comment:
At some point we may want to distinguish between optimistic data (`cache.extract(true)`) and realistic data (`cache.extract(false)`), so that people can see how their optimistic mutations are working.